### ISSUE TITLE
The Stars of Ifrit - key item on Airship - difficult, but not a bug?

### ISSUE THREAD

**Issue by Althaia** [EdenServer/community#2181]
### Checklist

* [x] I have searched for existing issues for issues like this one. The issue has not been posted. (Duplicate reports slow down development.)
* [x] I have provided reproducible steps. (No "as the title says" posts please. Provide reproducible steps even if it seems like a waste of time.)
* [x] I have provided my client version in the details. (type /ver into your game window)

### Details

My client version is: Version 30181205_0

I'm on "The Stars of Ifrit" quest (it's active in my Bastok - Current Quests section).

The guide says I should be able to get the key item from the ??? on the Sandoria-Jeuno Airship from 19:40 to 06:00 with moon ≥ 90%.

First Attempt:

1. I tried on Darksday with Full Moon (93%) at approx. 4:20 and clicking on the ??? didn't do anything. I tried many times.
2. I boarded from San d'Oria and landed in Jeuno.

Second Attempt:

1. I tried on Darksday with Full Moon (93%) at approx. 19:25 (my flight started at 19:20); it didn't work then, which makes sense if night doesn't start till 19:40...
2. I was busy writing this bug report, so I didn't try again until 20:00. It worked at that time.
3. I boarded from Jeuno and landed in San d'Oria.

Is there a more narrow definition of Nighttime for this quest and/or Eden Server? Would be good to know if so.

To summarize, I'm not sure why it didn't work at 4:20 on the way from Sandy to Jeuno. Maybe it didn't count as night anymore? Or it doesn't work if you go Sandy-to-Jeuno, only when you go Jeuno-to-Sandy? I dunno.

But I've also provided an example of something that did work, so... not sure if it's a bug or the values for Night are more restricted on this server than in the guide I was using. ;P

**Comment by Deepupuu** [comment]
93% full moon 22:00 in game on the sandy to jeuno ship i was able to get the KI so maybe it's fixed now?

**Comment by rwaffle** [comment]
Adding some additional info to this ticket, after investigating this quest.

Not sure if this has ever been changed from DSP, but the base DSP code for this quest calls for the time of day to be night to interact with the ???.

'if (TOTD == dsp.time.NIGHT and IsMoonFull()) then'

Based on what I can tell, time of day is split into smaller chunks of hours than what it potentially should be in game. For example, Night should be from 18:00-06:00, but it might be only coded as 20:00-00:00. Once the clock strikes 00:00, the time of day then becomes Midnight. Then at 04:00, time of day switches for Newday until 06:00. While these are all true, they should still fall under the designation of Night.

(From vana_time.cpp)

`<!--StartFragment-->    if (m_vMin == 0)    {        switch (m_vHour)        {            case  0: m_TimeType = TIME_NIGHT;   return TIME_MIDNIGHT;            case  4: m_TimeType = TIME_NEWDAY;  return TIME_NEWDAY;            case  6: m_TimeType = TIME_DAWN;    return TIME_DAWN;            case  7: m_TimeType = TIME_DAY;     return TIME_DAY;            case 17: m_TimeType = TIME_DUSK;    return TIME_DUSK;            case 18: m_TimeType = TIME_EVENING; return TIME_EVENING;            case 20: m_TimeType = TIME_NIGHT;   return TIME_NIGHT;        }    }    return TIME_NONE;<!--EndFragment-->`

Looking at some other items where time is a factor, you can see from mixins/families/ziz.lua that in order for the conditiions to span from 20:00 through 04:00, the code is written to be Night OR Midnight.

`if totd == dsp.time.NIGHT or totd == dsp.time.MIDNIGHT then -- 20:00 to 4:00`

Other scripts appear to handle this issue by using VanadielHour() instead of VandielTOTD(), such as:

`elseif (VanadielHour() >= 16 or VanadielHour() <= 6) then`

All of this to say, that the original posters scenario of failing to complete the quest prior to 20:00 and after 00:00, while Deepupuu succeeded at 22:00 would support the above.

**Comment by 59blargedy** [comment]
resolved in June 28th, 2021 update
